app/models/smart_proxy.rb

```diff
@@ -88,12 +86,19 @@ def has_feature?(feature)
     self.features.any? { |proxy_feature| proxy_feature.name == feature }
   end
 
-  private
-
   def proxy_status
-    @proxy_status ||= ProxyStatus.new(self)
+    @proxy_status ||= { :version => ProxyStatus::Version.new(self) }
+    features.map(&:name).each do |feature|
+      normalized_feature_name = feature.delete(' ')
+      status = ProxyStatus.find_status_by_humanized_name(normalized_feature_name)
```

Comment:
Normalized name can not be downcased, as it is passed later as a class name. I can downcase it here and the upcase it on `ProxyStatus::Base` but it seems un-necessary